base/ltcmdhooks.dtx

```diff
@@ -255,6 +255,25 @@
 % part of the code.\footnote{We might change this behavior in the main
 % document slightly after gaining some usage experience.}
 %
+% \subsection{Command copies}
+%
+% Once a hook has been added to a command, it will be present in any copies
+% of that command which are made. For example, if in the preamble we have
+% \begin{verbatim}
+%   \NewDocumentCommand\foo{}{}
+%   \AddToHook{cmd/foo/after}{1}%
+% \end{verbatim}
+% then after |\begin{document}|, any use of \cs{NewCommandCopy} will
+% include the hook in the copy, for example
+% \begin{verbatim}
+%   \NewCommandCopy\baz\foo
+% \end{verbatim}
+% would include the hook in this copied command.
+%
+% \subsection{Grouping}
+%
+% Command hooks are added respecting \TeX{} grouping, such that any hooks
+% defined inside a group will only apply within that group.
```

Comment:
This is wrong. Patching a command (adding `\UseHookWithArguments...` to a command body) _is_ global.

```tex
\documentclass{article}
\begin{document}
\def\foo{foo}
{
  \AddToHook{cmd/foo/after}{[after]}
  \foo % "foo[after]"
}
\foo % "foo[after]"
\end{document}
```

From source code, `\__hook_def_cmd:w` is let to one of `\xdef` or `\gdef`, both act global.
https://github.com/latex3/latex2e/blob/7c825d41be4fab5f0fdd7fd18c6d180febc2d41e/base/ltcmdhooks.dtx#L1026-L1044